src/dlgTriggerEditor.h

```diff
@@ -179,13 +179,15 @@ class dlgTriggerEditor : public QMainWindow, private Ui::trigger_editor
     void show_vars();
     void setThemeAndOtherSettings(const QString&);
 
-    static const int cmTriggerView;
-    static const int cmTimerView;
-    static const int cmAliasView;
-    static const int cmScriptView;
-    static const int cmActionView;
-    static const int cmKeysView;
-    static const int cmVarsView;
+    enum class EditorViewType {
```

Comment:
Look at the `enum SearchDataRole`, `enum SearchDataResultType` and `enum SearchOption` in the same file - I would suggest changing this `enum` to:

    enum EditorView {
        // Unset:
        ViewUnknown = 0x0,
        ViewTriggers = 0x1,
        ViewTimers = 0x2,
        ViewAliases = 0x4,
        ViewScripts = 0x8,
        ViewButtons = 0x10,
        ViewKeys = 0x20,
        ViewVariables = 0x04
    };

but note that the first two use a 0,1,2,3... value sequence as only one option can be used at a time.  Contrast that with the third which could eventually be using a 0,1,2,4... value sequence AND has a `Q_DECLARE_FLAGS(SearchOptions,SearchOption)` indicating that the values can be combined {similarly I would suggest `Q_DECLARE_FLAGS(EditorViews,EditorView)` for this case and also spot the `Q_DECLARE_OPERATORS_FOR_FLAGS(dlgTriggerEditor::EditorViews)` at the bottom of the header file OUTSIDE of the `class` declaration - which means that we can use `operator!()`/`operator&()`/`operator|()` to NOT/bitwise-AND/bitwise-OR on an `EditorViews` type variable (which will be when we change `(int) mCurrentView` to `(EditorViews) currentView` in the header file as well.  The reason for these extra bits is so that we can do tests like:

    if (mCurrentView & ViewTriggers) {
        ....
    }

in the `dlgTriggerEditor` file and if the `public`/`private` stuff is done right and we need to refer to an `EditorViews` value externally we just add the classname i.e. given a:

    EditorView getCurrentView() const {return mCurrentView;}

one-liner getter in the header file, another class will be able to check what we are currently showing and to use it in a test, e.g. something like:

    if (mpHost->mpEditorDialog->getCurrentView() & (~ dlgTriggerEditor::ViewButtons)) {
        ....
    }

to be able to tell that we have one of the views showing that is NOT the buttons/menus/toolbars one {I need a test like that if I move the CSS entry widget out of the rest of the widget that displays the details of those type of items into a separate widget that can be resized independently when I sort out the `QSplitter` container widget used on the right hand side of the editor which is currently a bit limited in what can be adjusted for different screen sizes.

Note that this is using the Qt `QFlag`/`QFlags` classes and that there is no `class` word in the `enum` this means that the underlying type is an `signed` or `unsigned` `int` which is an older way of doing things but will work better if we do away with the ridiculous set of `booleans` in the `TTreeWidget` class:

    bool mIsTriggerTree;
    bool mIsAliasTree;
    bool mIsScriptTree;
    bool mIsTimerTree;
    bool mIsKeyTree;
    bool mIsVarTree;
    bool mIsActionTree;

which could be simply replaced by another `enum` which could have a one-to-one mapping to this one (same underlying values) so one could be `static_cast` to the other in a quick and dirty manner should it be usefulto do so. 👿

Reply:
Um, that might be a bit TMI but does it give you any pointers as to what I think can be done?